packages/reason-relay/src/ReasonRelay.re

```diff
@@ -503,39 +503,22 @@ module Environment = {
   [@bs.send] external getStore: t => Store.t = "getStore";
 };
 
-module Context = {
-  type t;
-
-  type contextShape = {. "environment": Environment.t};
-
-  [@bs.module "react-relay"]
-  external context: React.Context.t(option(contextShape)) =
-    "ReactRelayContext";
-  let provider = React.Context.provider(context);
+module EnvironmentProvider = {
+  [@bs.module "relay-experimental"] [@react.component]
+  external make:
+    (~environment: Environment.t, ~children: React.element) => React.element =
+    "RelayEnvironmentProvider";
 
-  module Provider = {
-    [@react.component]
-    let make = (~environment: Environment.t, ~children) =>
-      React.createElement(
-        provider,
-        {"value": Some({"environment": environment}), "children": children},
-      );
-  };
+  let make = make;
 };
 
 let useConvertedValue = (convert, v) =>
   React.useMemo1(() => convert(v), [|v|]);
 
 exception EnvironmentNotFoundInContext;
 
-let useEnvironmentFromContext = () => {
-  let context = React.useContext(Context.context);
-
-  switch (context) {
-  | Some(ctx) => ctx##environment
-  | None => raise(EnvironmentNotFoundInContext)
-  };
-};
+[@bs.module "relay-experimental"]
+external useRelayEnvironment: unit => Environment.t = "useRelayEnvironment";
```

Comment:
Hmm, I'd rather not rename these at this point (`Context.Provider` and `useEnvironmentFromContext`).

Reply:
Alright, I will revert